src/mqt/bench/cli.py

```diff
@@ -23,56 +26,127 @@ def format_help(self) -> str:
+        "--target-directory",
+        type=str,
+        default=".",
+        help="Directory to save the output file (only used for 'qpy' or if --save is specified).",
+    )
+    parser.add_argument(
+        "--target-filename",
+        type=str,
+        default="",
+        help="Base filename (without extension) for saved output.",
+    )
+    parser.add_argument(
+        "--save",
+        action="store_true",
+        help="If set, save the output to a file instead of printing to stdout (required for 'qpy').",
+    )
+
     args = parser.parse_args()
 
+    # Build Qiskit settings
     qiskit_settings = QiskitSettings()
     if args.qiskit_optimization_level is not None:
         qiskit_settings = QiskitSettings(args.qiskit_optimization_level)
 
-    # Note: Assertions about argument validity are in get_benchmark()
-
-    # Temporary workaround to "get things working" with benchmark instances.
-    # This special treatment should be removed as soon as instance handling has been refactored in get_benchmark().
+    # Parse algorithm and optional instance
     benchmark_name, benchmark_instance = parse_benchmark_name_and_instance(args.algorithm)
 
-    result = get_benchmark(
+    # Generate circuit
+    circuit = get_benchmark(
         benchmark_name=benchmark_name,
         benchmark_instance_name=benchmark_instance,
         level=args.level,
         circuit_size=args.num_qubits,
-        compiler_settings=CompilerSettings(
-            qiskit=qiskit_settings,
-        ),
-        provider_name=args.native_gate_set,
-        device_name=args.device,
+        compiler_settings=CompilerSettings(qiskit=qiskit_settings),
+        provider_name=args.native_gate_set or "ibm",
+        device_name=args.device or "ibm_washington",
     )
-    print(qiskit_circuit_to_str(result))
-    return
 
+    try:
+        fmt = OutputFormat(args.output_format)
+    except ValueError:
+        msg = f"Unknown output format: {args.output_format}"
+        raise ValueError(msg) from None
+
+    # For QASM outputs, serialize and print
+    if fmt in (OutputFormat.QASM2, OutputFormat.QASM3) and not args.save:
+        header = generate_header(fmt)
+        serial = qasm2_dumps(circuit) if fmt == OutputFormat.QASM2 else qasm3_dumps(circuit)
+        sys.stdout.write(header)
+        sys.stdout.write(serial)
+        return
+
+    # Otherwise, save to file
+    filename = args.target_filename or f"{benchmark_name}_{args.level}_{args.num_qubits}_{fmt.value}"
+    success = save_circuit(
+        qc=circuit,
+        filename=filename,
+        output_format=str(fmt.value),
+        target_directory=args.target_directory,
+    )
+    if not success:
+        sys.exit(1)
+    # Optionally, inform user of file location if saving
+    if args.save or fmt == OutputFormat.QPY:
+        ext = "qasm" if fmt in (OutputFormat.QASM2, OutputFormat.QASM3) else fmt.value
+        path = Path(args.target_directory) / f"{filename}.{ext}"
+        print(path)
```

Comment:
It would be nice to not replicate too much logic here.
Can we somehow centralise this in the `output.py` file?
This is not so different to writing to a file. It just writes to stdout instead.